src/plugins/github/graphql.js

```diff
@@ -340,6 +340,108 @@ function* continuationsFromReview(
   }
 }
 
+/**
+ * Merge structured data from the given `source` into a given subpath of
+ * the `destination`. The original inputs are not modified.
+ *
+ * Arrays are merged by concatenation. Objects are merged by recursively
+ * merging each key. Primitives are merged by replacement (the
+ * destination is simply overwritten with the source).
+ *
+ * See test cases for examples.
+ *
+ * NOTE: The type of `source` should be the same as the type of
+ *
+ *     destination[path[0]][path[1]]...[path[path.length - 1]],
+ *
+ * but this constraint cannot be expressed in Flow so we just use `any`.
+ */
+export function merge<T>(
```

Comment:
complaint: i find this function a little difficult to parse; it's almost 100 lines of code with inline functions but without any paragraph breaks. breaking it up visually (e.g. on the boundaries of inline functions) or breaking it up semantically (into mergeArray or mergeObject as appropriate) could be helpful.

Reply:
Addressed in https://github.com/sourcecred/sourcecred/pull/122#issuecomment-378859005.